### PR TITLE
Pin mypy version

### PR DESCRIPTION
**Pull request by Czaki**
# Description

This PR pin mypy version is used for test typing and enabling dependaboot to create PR with updates of mypy related things. 

This will allow us to not get unrelated failures in PR. 

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Review by kephale — APPROVED** [review]
Awesome, @Czaki!

File: resources/requirements_mypy.txt
Comment (Czaki) [inline resources/requirements_mypy.txt:3]: I see benefits from pin `mypy` but I'm not sure if two others also should be pinned. On one side it reduces the probability of random error, on the other side, I do not remember errors with this packages. Maybe release this versions to reduce dependaboot PRs?

Comment (kephale) [reply resources/requirements_mypy.txt:3]: I agree and vote to pin them. There is a strong advantage to having consistent type checking, so it is nice to make sure these packages don't change unexpectedly.

Comment (brisvag) [reply resources/requirements_mypy.txt:3]: Yeah I agree, I don't mind a few extra PRs if it means we don't get spurious unexpected failures.

**Review by brisvag — APPROVED** [review]

File: .github/dependabot.yml
Comment (jni) [inline .github/dependabot.yml]: I think monthly is a nicer experience here?

Comment (Czaki) [reply .github/dependabot.yml]: I prefer fast information about problems. Maybe weekly. But this is a weak opinion.

Comment (kephale) [reply .github/dependabot.yml]: I'll vote for `weekly`. `daily` is fast enough that I would ignore it, but a month could be a long time to not know about a vulnerability.

Comment (jni) [reply .github/dependabot.yml]: My personal opinion is that we have a very high volume of issues and PRs, and we want to reduce that as much as possible. Since the dependencies are pinned, I don't see a huge advantage in getting notifications early. Even yearly would be fine. However, I'm generally unlikely to be the one dealing with these so I defer to those that will. 🤷 My opinion is also weak, hence why I didn't block or anything — people should feel free to merge this when ready! 😊

**Pull request merged**
